Source/Manager.swift

```diff
@@ -9,144 +9,159 @@
-            resources[resource.resourceIdentifier] = resource
+      // if it does...
+      if let id = Int(component) {
+
+        // add it to the IDs array to be returned
+        if ids?.append(id) == nil {
+          ids = [id]
         }
-    }
 
-    /// Remove a resource from the manager
-    static public func removeResource(resource: ResourceProtocol) {
-        resources.remove(resource.resourceIdentifier)
+        // remove the number and replace with predetermined key to be used for the name to be returned
+        pathComponents[index] = ":id"
+      }
     }
 
-    /// Remove all resources from the manager
-    static public func removeResources() {
-        resources.removeAll()
+    // prevents "//" from occurring when joining the path components
+    if pathComponents.first == separator {
+      pathComponents.removeFirst()
     }
-    
-    // MARK: - Helpers
-    
-    /// Normalize a path which can be used as a Resource Identifier and requested resource IDs, if available.
-    /// - Returns: "/path/to/resource/123?foo=bar&baz=quux" -> ("/path/to/resource/", 123, ["foo": ["bar"], "baz": ["quux"])
-    static func resourceArtifactsFromURL(url: NSURL) -> (name: NameArtifact, ids: IDArtifacts?, query: QueryArtifacts?)? {
-        guard var pathComponents = url.pathComponents else { return nil }
-
-        var name: String?
-        var ids = [Int]?()
-        let separator = "/"
-
-        // Check if the URL has an ID within it -> /api/path/to/123/example
-        for (index, component) in pathComponents.enumerate() {
-
-            // if it does...
-            if let id = Int(component) {
-
-                // add it to the IDs array to be returned
-                if ids?.append(id) == nil {
-                    ids = [id]
-                }
-                
-                // remove the number and replace with predetermined key to be used for the name to be returned
-                pathComponents[index] = ":id"
-            }
-        }
 
-        // prevents "//" from occurring when joining the path components
-        if pathComponents.first == separator {
-            pathComponents.removeFirst()
-        }
-        
-        name = pathComponents.joinWithSeparator(separator)
-        name = separator + name!
+    name = pathComponents.joinWithSeparator(separator)
+    name = separator + name!
 
-        let query = queryDictionaryFromURL(url)
-        
-        return (name!, ids, query)
+    let body = ArtifactDictionaryFromData(request.HTTPBody)
+    let query = ArtifactDictionaryFromString(url.query)
+
+    return (name!, ids, query, body)
+  }
+
+  /// Return a dictionary with arrays populated with the values of HTTP body data. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromData(data: NSData?) -> QueryArtifacts? {
+    guard let data = data,
+      let componentString = String(data: data, encoding: NSUTF8StringEncoding)
```

Comment:
I asked the community at large this same question. I provided a few possibilities in my post but where we ended up in conversation was that Xcode tries to hassle you like a bully and it can be annoying/hard to format things the way you want to.  Here's a snippet of their response:

![screen shot 2016-02-04 at 3 07 52 pm](https://cloud.githubusercontent.com/assets/1699437/12829728/3b8261b4-cb51-11e5-9ae0-e860a1b615e2.png)

But in my opinion I like the style that you posted. What do you think?